application/models/SurveyDeactivator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Service class to deactivate survey.
+ * @todo Move to models/services/survey/ folder.
+ */
+
+class SurveyDeactivator
+{
+    /** @var Survey */
+    protected $survey;
+
+    /** @var array */
+    protected $tableDefinition = [];
+
+    /** @var array */
+    protected $timingsTableDefinition = [];
+
+    /** @var string */
+    protected $error;
+
+    public function __construct($survey = null)
+    {
+        $this->$survey = $survey;
```

Comment:
a bug here, should be $this->s.., not $this->$s... Error 500 "Creation of dynamic property SurveyDeactivator::$ is deprecated"

Reply:
Thanks, @TonisOrmisson for pointing this out, created a PR to fix it